lib/httpadapter/httpadapter.js

```diff
@@ -55,14 +56,10 @@ HttpAdapter.prototype.get = function(url, params, callback) {
     //the whole response has been recieved, so we just print it out here
     response.on('end', function() {
       if (response.statusCode !== 200) {
-        return callback(new Error('Response status code is ' + response.statusCode), null);
+        return callback(new Error('Response status code is ' + response.statusCode), _this._formatContent(contentType, str));
```

Comment:
here instead of passing a second argument to the callback, I would add a property `response: {body: results, status: 200}` to the error object